packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -513,95 +531,122 @@ export function NativeStackView({ state, navigation, descriptors }: Props) {
 
   const { colors } = useTheme();
 
+  const currentRouteKey = state.routes[state.index].key;
+  const topScreenOptions = descriptors[currentRouteKey].options;
+  const gestureDetectorBridge = React.useRef<GestureDetectorBridge>({
+    stackUseEffectCallback: (_stackRef) => {
+      // this method will be overriden in GestureDetector
+    },
+  });
+  type RefHolder = Record<
+    string,
+    React.MutableRefObject<React.Ref<NativeStackNavigatorProps>>
+  >;
```

Comment:
Normally we define types at the top level of the file. Though seems this type is also defined in `types.tsx` and could be imported from there.

Reply:
Done in https://github.com/react-navigation/react-navigation/pull/11943/commits/301cd8dace25c46ed8b0f5931333176b07dcc99c.